packages/vision/src/helpers.js

```diff
@@ -73,14 +73,27 @@ var coerceImage = (image, callback) => {
  * @return {Function} - The function that, when called, will call annotateImage
  *   asking for the single feature annotation.
  */
-var _createSingleFeatureMethod = featureValue => {
-  return function(image, options) {
-    return this.annotateImage({
-      image: image,
-      features: [{type: featureValue}],
-    }, options);
-  };
-};
+ var _createSingleFeatureMethod = featureValue => {
+   return function(annotateImageRequest, callOptions) {
+     annotateImageRequest.features = annotateImageRequest.features || [];
+
+     // Ensure the feature value indicated by the user's method choice exists on the
+     // features array; if it does not, add it.
+     var found = false;
```

Comment:
I think we should only allow the one `featureValue` to be the feature on the request. So basically, that would mean overriding the user's input, because I think it would be a misuse of our API if they did something like this:

```js
cropHints({
  image: {...},
  features: ['labeldetection', 'crophints']
})
```

Reply:
If it is incorrect, then we should throw an exception. We should not just plow over half the input.

Reply:
I would have preferred @jmdobry's solution initially, but typed and backspaced it because it didn't seem worth the challenge of bringing you down from your position. I wanted to make sure all of the options were presented, so that your perspective would be a contribution to the best decision. I was only strongly opposed to having consecutive, optional arguments that would not work as advertised.

You are the charging force that moves us forward, and as this is a mostly-generated API that you developed, I respect your decision. Let's roll with the throwing thing 👍

Reply:
It is _always_ fine to challenge me. :-)

I think we might be at an impasse though. Let's make sure @jmdobry does not strongly disagree before making a final call.